azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -284,22 +296,25 @@ public Set<Integer> getContainersOfTerminatedFlows() {
   }
 
   /**
-   * Get the flows of EXECUTION_STOPPED status
+   * Get the flows recently (last 15 minutes) terminated: status in (execution_stopped, killed,
+   * killing, failed, failed_finishing, failed_succeeded, canceled)
    */
   @NotNull
-  Set<Integer> getExecutionStoppedFlows() {
-    Set<Integer> executionStoppedFlows = new HashSet<>();
-    if (this.executionStoppedMap.containsKey(Status.EXECUTION_STOPPED)) {
+  Set<Integer> getRecentTerminationFlows() {
+    Set<Integer> recentKilledFlows = new HashSet<>();
+    this.recentTerminatedStatusMap.forEach((status, value) -> {
       try {
         List<ExecutableFlow> flows = this.executorLoader.fetchFreshFlowsForStatus(
-            Status.EXECUTION_STOPPED, this.executionStoppedMap);
-        executionStoppedFlows.addAll(
+            status, this.recentTerminatedStatusMap);
+        recentKilledFlows.addAll(
             flows.stream().map(ExecutableFlow::getExecutionId).collect(Collectors.toSet()));
+        logger.error("Got recent termination flows executions of Status " + status + ": " +
```

Comment:
Should this be logged as an 'error' or an info/warn ?
What's the expected number of such flows (worst case observation) ?

Reply:
1. Yes it should be `info`
2. Will investigate. In our test cluster I think it's <30 in total